pkg/provider/acceptance_grant_table_test.go

```diff
@@ -0,0 +1,137 @@
+	"testing"
+
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/materialize"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/jmoiron/sqlx"
+)
+
+func TestAccGrantTable_basic(t *testing.T) {
+	privilege := randomPrivilege("TABLE")
+	roleName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	tableName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	schemaName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	databaseName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGrantTableResource(roleName, tableName, schemaName, databaseName, privilege),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGrantTableExists("materialize_grant_table.table_grant", roleName, tableName, schemaName, databaseName, privilege),
+					resource.TestCheckResourceAttr("materialize_grant_table.table_grant", "role_name", roleName),
+					resource.TestCheckResourceAttr("materialize_grant_table.table_grant", "privilege", privilege),
+					resource.TestCheckResourceAttr("materialize_grant_table.table_grant", "table_name", tableName),
+					resource.TestCheckResourceAttr("materialize_grant_table.table_grant", "schema_name", schemaName),
+					resource.TestCheckResourceAttr("materialize_grant_table.table_grant", "database_name", databaseName),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGrantTable_disappears(t *testing.T) {
+	privilege := randomPrivilege("TABLE")
+	roleName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	tableName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	schemaName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	databaseName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGrantTableResource(roleName, tableName, schemaName, databaseName, privilege),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGrantTableExists("materialize_grant_table.table_grant", roleName, tableName, schemaName, databaseName, privilege),
+					testAccCheckGrantTableRevoked(roleName, tableName, schemaName, databaseName, privilege),
+				),
+				PlanOnly:           true,
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccGrantTableResource(roleName, tableName, schemaName, databaseName, privilege string) string {
+	return fmt.Sprintf(`
+resource "materialize_role" "test" {
+	name = "%s"
+}
+
+resource "materialize_database" "test" {
```

Comment:
Not sure about including database and schema resources for these tests when we are only really after the table grants. Mostly doing it because the scan id functions require all the qualifiers and would otherwise require hardcoding.